lib/public/utilities/ObservableData.js

```diff
@@ -27,20 +27,110 @@ export class ObservableData extends Observable {
         this._current = initial;
     }
 
+    /**
+     * Returns a builder to create a new observable data
+     *
+     * @return {ObservableDataBuilder} the builder
+     */
+    static builder() {
+        return new ObservableDataBuilder();
+    }
+
     /**
      * Returns the current value of the data
      * @return {T} the current value
      */
-    get current() {
+    getCurrent() {
         return this._current;
     }
 
     /**
      * Set the current value of the data and notify
      * @param {T} value the current value
+     * @return {void}
      */
-    set current(value) {
+    setCurrent(value) {
         this._current = value;
         this.notify();
     }
 }
+
+/**
+ * Observable data that applies a processing on every value that is pushed into it
+ */
+class ObservableDataWithProcessing extends ObservableData {
+    /**
+     * Constructor
+     *
+     * @param {initialValue} initialValue the initial value of the observable data
+     * @param {function} processors the list of processing to apply to the values (processors must be chainable)
+     */
+    constructor(initialValue, processors) {
+        super(initialValue);
+
+        this._processors = processors;
+    }
+
+    // eslint-disable-next-line valid-jsdoc
+    /**
+     * @inheritDoc
+     */
+    setCurrent(value) {
+        let processedValue = value;
+
+        for (const processor of this._processors) {
+            processedValue = processor(processedValue);
+        }
+
+        super.setCurrent(processedValue);
+    }
+}
+
+/**
+ * Builder to create instance of observable data
+ *
+ * @template T
+ */
+class ObservableDataBuilder {
+    /**
+     * Constructor
+     */
+    constructor() {
+        this._initialValue = null;
+        this._processors = [];
+    }
+
+    /**
+     * Set the initial value of the observable data
+     *
+     * @param {T} initialValue the initial value of the observable data
+     * @return {this} the builder instance
+     */
+    initialValue(initialValue) {
+        this._initialValue = initialValue;
+        return this;
+    }
+
+    /**
+     * Adds a processing function that will be applied to the observable data (map functions will be called in the same
+     * order as they have been registered)
+     *
+     * @param {function} processor the function to apply to observable data
+     * @return {this} the builder instance
+     */
+    map(processor) {
```

Comment:
I think map is more concise and still makes sense if you think of the way you build it, but I may be biaised.
```js
createBuilder()
    .initialValue()
    .map() // map the value
    .map() // map the value
    .build();
```
Another option would be to name it `pipe`
@graduta what do you think?

Reply:
Given the example above in which in the map you pass an action of mapping, perhaps `map` is not the most representative.
I would suggest something like `apply` as it would become:
```
createBuilder()
  .initialValue()
  .apply((remoteData) => remoteData.map({ Success: ({ data }) => data }))
  .build()
```

Reply:
Renamed to apply